models/DataObject/ClassDefinition/Data/Fieldcollections.php

```diff
@@ -686,8 +686,7 @@ public function classSaved($class, $params = [])
                     $fieldDefinition = $definition->getFieldDefinitions();
 
                     foreach ($fieldDefinition as $fd) {
-                        //TODO Pimcore 11 remove method_exists call
-                        if (!$fd instanceof DataContainerAwareInterface && method_exists($fd, 'classSaved')) {
+                        if (!$fd instanceof DataContainerAwareInterface && $fd instanceof ClassSavedInterface) {
```

Comment:
BC Break, please add a BC layer
```suggestion
                        if (!$fd instanceof DataContainerAwareInterface && ($fd instanceof ClassSavedInterface || method_exists($fd, 'classSaved'))) {
                            if (!$fieldDefinition instanceof ClassSavedInterface) {
                                trigger_deprecation('pimcore/pimcore', '10.6',
                                    sprintf('Usage of method_exists is deprecated since version 10.6 and will be removed in Pimcore 11.' .
                                    'Implement the %s interface instead.', ClassSavedInterface::class));
                            }
```
Also please add a Update Note.

Reply:
Why should this be a BC break? each class implements the interface and the corresponding method ...

Reply:
Every class in this repro. But devs can create own data types.

Reply:
Your right - will provide a BC layer for that 👍

Reply:
@Corepex Please also create a follow-up issue for `11.x` to remove the BC layer 😉